### PR TITLE
[RISCV64] Support WasmEdge on RISC-V64 arch.

### PR DESCRIPTION
**Pull request by O3Ol**
# Motivation

This pull request is to add files which make WasmEdge build for `RISC-V 64` arch.

# Details

- Add the definitions of riscv64.
- Add `build_for_riscv.yml` for CI test.
- Add the doc `riscv64.md` for buiding WasmEdge on RISC-V64.

### CODE REVIEW

**Comment by hydai** [comment]
Hi @O3Ol 
Please fix the failing CI. Thanks.

File: lib/aot/compiler.cpp [hunk 1]
```diff
@@ -5146,6 +5149,9 @@ Expect<void> Compiler::compile(Span<const Byte> Data, const AST::Module &Module,
     if (!Conf.getCompilerConfigure().isGenericBinary()) {
       CPUName = llvm::sys::getHostCPUName();
     }
+#if defined(__riscv) && __riscv_xlen == 64
+    CPUName = "generic-rv64";
```
Comment (hydai) [inline lib/aot/compiler.cpp:5153]: Just want to know if `llvm::sys::getHostCPUName()` will return `generic-rv64` or not.
I think this should be moved to line 5145. And you can create a constant string `"generic"` or `"generic-rv64`, depends on the platform.

File: .github/workflows/build_for_riscv.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    needs: lint
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: uraimo/run-on-arch-action@v2
+      name: Build WasmEdge
+      id: build
+      with:
+        arch: none
```
Comment (hydai) [inline .github/workflows/build_for_riscv.yml:58]: Do we need to specify the `rv64` here?

**Review by hydai — CHANGES_REQUESTED** [review]

File: lib/aot/compiler.cpp (hunk 1 above)
Comment (O3Ol) [reply lib/aot/compiler.cpp:5153]: In my environment `llvm::sys::getHostCPUName()` did not return `generic-rv64`, so I defined it. I would move it to line 5145 as you said.

File: .github/workflows/build_for_riscv.yml (hunk 2 above)
Comment (O3Ol) [reply .github/workflows/build_for_riscv.yml:58]: No, I will remove unnecessary content.

**Review by hydai — APPROVED** [review]

**Comment by hydai** [comment]
@O3Ol Thanks for your contribution!

**Pull request merged**